uspace/lib/cpp/include/__bits/thread/packaged_task.hpp

```diff
@@ -0,0 +1,228 @@
+/*
+ * Copyright (c) 2019 Jaroslav Jindrak
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * - Redistributions of source code must retain the above copyright
+ *   notice, this list of conditions and the following disclaimer.
+ * - Redistributions in binary form must reproduce the above copyright
+ *   notice, this list of conditions and the following disclaimer in the
+ *   documentation and/or other materials provided with the distribution.
+ * - The name of the author may not be used to endorse or promote products
+ *   derived from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE AUTHOR ``AS IS'' AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+ * OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT
+ * NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+ * THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef LIBCPP_BITS_THREAD_PACKAGED_TASK
+#define LIBCPP_BITS_THREAD_PACKAGED_TASK
+
+#include <__bits/exception.hpp>
+#include <__bits/functional/function.hpp>
+#include <__bits/memory/allocator_traits.hpp>
+#include <__bits/thread/future.hpp>
+#include <__bits/thread/future_common.hpp>
+#include <__bits/thread/shared_state.hpp>
+#include <type_traits>
+#include <utility>
+
+namespace std
+{
+    /**
+     * 30.6.9, class template packaged_task:
+     */
+
+    template<class>
+    class packaged_task; // undefined
```

Comment:
What does "undefined" mean here?

Reply:
It means that the base template does not exist and only explicitly created specializations can be used. In this case it is used to require the template parameter to be a function/callable, denoted by R(Args...) for a given return type R and argument types Args...

Reply:
Should probably be explained in the comment. Just the word "undefined" doesn't tell me how or why. :)

Reply:
Doesn't it? It does right what it says on the box, that template is not defined ;) But sure, I can add a comment.

Reply:
Actually, I think that exact comment is taken straight from the standard (but still, I will change it).